envoy-filters/mgw-source/filters/http/mgw-wasm-websocket/filter.cc

```diff
@@ -129,36 +175,38 @@ FilterDataStatus MgwWebSocketContext::onRequestBody(size_t body_buffer_length,
     // Read ext_authz_metadata_ metdata saved as a member variable
     *request.mutable_metadata() = *this->metadata_;
     request.set_payload(std::string(body->view()));
+    request.set_direction(WebSocketFrameRequest_MessageDirection_PUBLISH);
     
     // Perform throttling logic.
     // If the throttle state is underlimit and if the gRPC stream is open, send WebSocketFrameRequest. 
     // If no gRPC stream, try to open a new stream and then send. 
     if(this->throttle_state_ == ThrottleState::UnderLimit){
-      if(this->handler_state_ == HandlerState::OK){
-        LOG_TRACE(std::string("gRPC bidi stream available. publishing frame data..."));
-        auto ack = this->stream_handler_->send(request, false);
-        if (ack != WasmResult::Ok) {
-          LOG_TRACE(std::string("error sending frame data")+ toString(ack));
-        }
-        LOG_TRACE(std::string("frame data successfully sent:"+ toString(ack)));
-      }else{
-        establishNewStream();
-        auto ack = this->stream_handler_->send(request, false);
-        if (ack != WasmResult::Ok) {
-          LOG_TRACE(std::string("error sending frame data")+ toString(ack));
-        }
-        LOG_TRACE(std::string("frame data successfully sent:"+ toString(ack)));
-      }
+      request.set_apim_error_code(0);
+      sendEnforcerRequest(this, request);
       return FilterDataStatus::Continue;
     // If throttle state is FailureModeAllowed, then try to esatblish a new gRPC stream and 
-    // pass the request to next filter. 
+    // pass the request to next filter. This state switch happens when the filter-enforcer connection fails.
     }else if (this->throttle_state_ == ThrottleState::FailureModeAllowed){
       establishNewStream();
+      request.set_apim_error_code(0);
+      auto ack = this->stream_handler_->send(request, false);
+      if (ack != WasmResult::Ok) {
+        LOG_WARN(std::string("error sending frame data ")+ toString(ack) + std::string(" : ") + this->x_request_id_);
+      }
+      LOG_TRACE(std::string("frame data successfully sent:" + toString(ack)) + std::string(" : ") +
+          this->x_request_id_);
```

Comment:
I think you have missed here. successful log gets printed even when result is not ok